game-core/src/main/java/games/strategy/triplea/ui/PickTerritoryAndUnitsPanel.java

```diff
@@ -154,8 +154,7 @@ public void actionPerformed(final ActionEvent event) {
         if (unitChoices.size() < unitsPerPick) {
           // if we have fewer units than the number we are supposed to pick, set it to all
           pickedUnits.addAll(unitChoices);
-        } else if (!unitChoices.isEmpty()
-            && unitChoices.stream().allMatch(Matches.unitIsOfType(unitChoices.get(0).getType()))) {
+        } else if (unitChoices.stream().allMatch(Matches.unitIsOfType(unitChoices.get(0).getType()))) {
```

Comment:
I must admit I'm not 100% sure about this one.
But I think intellij figured out that unitsPerPick must be 1 or greater, which means that if the collection is empty the if case before will be chosen instead.
I'd be OK reverting this if you are not convinced, but usually IntelliJ is pretty good with bool expressions

Reply:
Yeah, I think you're right.  It has something to do with determining that `unitsPerPick` is always greater than zero.  In that case, if `unitChoices` is empty, the first branch will be taken.

I followed the code backwards to where `unitPerPick` ultimately gets initialized from `TripleAPlayer#pickTerritoryAndUnits()`.  That method short-circuits if `unitsPerPick` is less than one, so as to guarantee that `PickTerritoryAndUnitsPanel#unitsPerPick` is always greater than zero.

So, I guess the change is good.  The only concern I have is someone coming back to this code in a year and being confused just like we were.

 Do you think adding a precondition check in `PickTerritoryAndUnitsPanel#waitForPickTerritoryAndUnits()` for `unitsPerPick > 0` would be helpful here?

Also, an additional comment below line 155 might be helpful, as well.  Something like "This branch will be taken if unitChoices is empty because unitsPerPick > 0".  But that's probably not as important as the precondition.  If you don't think the comment is useful, don't worry about it.